### PR TITLE
Make Sorbet faster to start in debug build

### PR DESCRIPTION
**Pull request by jez**
### Motivation

It had been starting to annoy me how long it would take to start Sorbet in debug
mode, so I went and profiled it and culled the biggest offenders.

### Test plan

Existing tests

```
~/sorbet (master)
❯ hyperfine --warmup 3 "bazel-bin/main/sorbet -e 0"
Benchmark 1: bazel-bin/main/sorbet -e 0
  Time (mean ± σ):     959.5 ms ±  13.4 ms    [User: 930.1 ms, System: 20.4 ms]
  Range (min … max):   939.5 ms … 985.7 ms    10 runs

~/sorbet (jez-timers)
❯ hyperfine --warmup 3 "bazel-bin/main/sorbet -e 0"
Benchmark 1: bazel-bin/main/sorbet -e 0
  Time (mean ± σ):     532.9 ms ±  15.0 ms    [User: 504.4 ms, System: 18.8 ms]
  Range (min … max):   520.3 ms … 567.2 ms    10 runs
```

### CODE REVIEW

File: core/GlobalState.cc [hunk 1]
```diff
@@ -2027,22 +2028,46 @@ void GlobalState::sanityCheck() const {
...
+        return;
+    }
+
+    Timer timeit(tracer(), "GlobalState::sanityCheck");
+
+    sanityCheckTableSizes();
+    sanityCheckNames();
+
     {
         Timer timeit(tracer(), "GlobalState::sanityCheck (symbols)");
```
Comment (neilparikh) [inline core/GlobalState.cc:2042]: Any reason not to split this off into `sanityCheckSymbols`? Do we always want to also do `sanityCheckTableSizes` and `sanityCheckNames` when this is done?

**Review by neilparikh — CHANGES_REQUESTED** [review]
You have a test failure

File: core/GlobalState.cc (hunk 1 above)
Comment (jez) [reply core/GlobalState.cc:2042]: I'm not quite sure what you're asking.

There were some places where I **only** wanted to sanity check the table sizes, or sanity check the names, so I factored out helpers for these.

There were no places where I **only** wanted to sanity check the symbols, so I did not factor them out of the top-level sanity check function.

Comment (neilparikh) [reply core/GlobalState.cc:2042]: Ah, that answers exactly what I was asking, thanks.

**Review by neilparikh — APPROVED** [review]

**Pull request merged**